app/com/arpnetworking/metrics/portal/scheduling/impl/JobScheduler.java

```diff
@@ -0,0 +1,225 @@
+/*
+ * Copyright 2018 Dropbox, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.arpnetworking.metrics.portal.scheduling.impl;
+
+import akka.actor.AbstractActorWithTimers;
+import akka.actor.ActorRef;
+import akka.actor.PoisonPill;
+import akka.actor.Props;
+import akka.pattern.PatternsCS;
+import com.arpnetworking.metrics.portal.scheduling.JobRef;
+import com.arpnetworking.steno.Logger;
+import com.arpnetworking.steno.LoggerFactory;
+import models.internal.scheduling.Job;
+import scala.Serializable;
+import scala.concurrent.duration.Duration;
+import scala.concurrent.duration.FiniteDuration;
+
+import java.time.Clock;
+import java.time.Instant;
+import java.time.temporal.ChronoUnit;
+import java.util.Optional;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * An actor that executes {@link Job}s.
+ *
+ * @param <T> The type of result produced by the {@link Job}s.
+ *
+ * @author Spencer Pearson (spencerpearson at dropbox dot com)
+ */
+public final class JobScheduler<T> extends AbstractActorWithTimers {
```

Comment:
Should this be called a `JobExecutorActor` or `ScheduledJobActor`? Seems potentially ambiguous with the `JobCoordinator` from the design proposal.